src/stcal/ramp_fitting/utils.py

```diff
@@ -1411,10 +1411,24 @@ def dq_compress_final(dq_int, n_int):
         combination of all integration's pixeldq arrays, 2-D flag
     """
     f_dq = dq_int[0, :, :]
+    nints = dq_int.shape[0]
 
     for jj in range(1, n_int):
         f_dq = np.bitwise_or(f_dq, dq_int[jj, :, :])
 
+    # Sum each pixel over all integrations where DO_NOT_USE is set.  If
+    # the number of integrations with DO_NOT_USE set is less than the
+    # total number of integrations, that means at least one integration
+    # has good data, so the final DQ flag for that pixel should NOT
+    # include the DO_NOT_USE flag.
+    dnu = np.zeros(dq_int.shape, dtype=np.uint32)
+    dnu[np.where(np.bitwise_and(dq_int, dnu_flag))] = 1
+    dnu_sum = dnu.sum(axis=0)
+    not_dnu = np.where(dnu_sum < nints)
+
+    not_dnu_flag = ~dnu_flag & 0xffffffff  # Forces it to be 32-bits
```

Comment:
would it also work to just add ".astype(uint32)" to "not_dnu_flag"?

Reply:
It doesn't appear that `.astype(uint32)` works for an integer, only for an `ndarray` or `pandas` data frame.  However, there is an `np.uint32()` function available.  I changed the code to use this function, instead of the cryptic `& 0xffffffff`.